common/core/desktop/src/kmx/kmx_xstring.cpp

```diff
@@ -215,7 +227,7 @@ int km::kbp::kmx::xstrpos(PKMX_WCHAR p1, PKMX_WCHAR p)
 PKMX_WCHAR km::kbp::kmx::xstrchr(PKMX_WCHAR buf, PKMX_WCHAR chr)
 {
   for(PKMX_WCHAR q = incxstr(buf); *buf; buf = q, q = incxstr(buf))
-    if(!u16ncmp(buf, chr, (int)(q-buf)))
+    if(!u16ncmp(buf, chr, (intptr_t)(q-buf)))
```

Comment:
Does the Windows version (line 167) remain the same?
```
    if(!wcsncmp(buf, chr, (int)(q-buf)))
```

Reply:
Good catch. I knew there was something else I needed to do. We should audit the rest of keyman32/keyman64 for use of improper casts.

It is unlikely to cause a problem presently because we currently have a base address fixed in the keyman64.dll below 2GB (which itself is not recommended...) so we should probably put this in as a maintenance issue. It could arise if Windows needs to relocate the dll I guess.